debug_toolbar/store.py

```diff
@@ -0,0 +1,127 @@
+import json
+from collections import defaultdict, deque
+from typing import Any, Dict, Iterable
+
+from django.core.serializers.json import DjangoJSONEncoder
+from django.utils.encoding import force_str
+from django.utils.module_loading import import_string
+
+from debug_toolbar import settings as dt_settings
+
+
+class DebugToolbarJSONEncoder(DjangoJSONEncoder):
+    def default(self, o: Any) -> Any:
+        try:
+            return super().default(o)
+        except TypeError:
+            return force_str(o)
+
+
+def serialize(data: Any) -> str:
+    return json.dumps(data, cls=DebugToolbarJSONEncoder)
+
+
+def deserialize(data: str) -> Any:
+    return json.loads(data)
+
+
+class BaseStore:
+    _config = dt_settings.get_config().copy()
+
+    @classmethod
+    def ids(cls) -> Iterable:
```

Comment:
Maybe request_ids to make the name a little bit less generic?

Reply:
That's a good idea.